File: src/main/java/com/yoghurt1131/weatherapi/application/service/WeatherInterpreter.java
package com.yoghurt1131.weatherapi.application.service;

import com.yoghurt1131.weatherapi.domain.input.valueobject.RangedWeather;
import com.yoghurt1131.weatherapi.domain.output.valueobject.Forecast;

import java.util.List;

/**
 * Apiレスポンスをもとに天気予報を作成する
 */
public interface WeatherInterpreter {

    Forecast interpret(List<RangedWeather> weatherData);
}

File: src/main/kotlin/dev/yoghurt1131/weatherapi/service/WeatherInterpreter.kt
package dev.yoghurt1131.weatherapi.service

import dev.yoghurt1131.weatherapi.domain.input.valueobject.RangedWeather
import dev.yoghurt1131.weatherapi.domain.output.valueobject.Forecast

/**
 * Apiレスポンスをもとに天気予報を作成する
 */
interface WeatherInterpreter {
    fun interpret(weatherData: List<RangedWeather>): Forecast;
}